src/transformers/models/maskformer/modeling_maskformer.py

```diff
@@ -2193,17 +1374,26 @@ def __init__(self, config: MaskFormerConfig):
                 The configuration used to instantiate this model.
         """
         super().__init__()
-        self.encoder = MaskFormerSwinTransformerBackbone(config.backbone_config)
+
+        # TODD: add method to load pretrained weights of backbone
+        backbone_config = config.backbone_config
+        if isinstance(backbone_config, SwinConfig):
```

Comment:
Can we check on the model type if this is the only place `SwinConfig` is used? (If not don't worry about it).
This would completely decouple MaskFormer from Swin.